File: app/src/main/java/com/bignerdranch/stockwatcher/AppComponent.java
package com.bignerdranch.stockwatcher;

import com.bignerdranch.stockwatcher.ui.StockInfoFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {
    void inject(StockInfoFragment fragment);
}

File: app/src/main/java/com/bignerdranch/stockwatcher/AppComponent.kt
package com.bignerdranch.stockwatcher

import com.bignerdranch.stockwatcher.ui.StockInfoFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun inject(fragment: StockInfoFragment)
}
